mysql-test/suite/rpl/t/rpl_enable_event_cmd_line_validation.test

```diff
@@ -0,0 +1,94 @@
+# === Purpose ===
+#
+# Test to verify that server config value assigned to --replica-enable-event
+# is properly handled during server startup.
+#
+# === References ===
+#
+# PS-6002: Create/alter event state to keep enabled on replicas
+
+# Test in this file is binlog format agnostic, thus no need
+# to rerun it for every format.
+--source include/big_test.inc
+--source include/have_binlog_format_row.inc
+--source include/master-slave.inc
+--source include/rpl_connection_slave.inc
+
+# Shut down the replica server.
+--let $rpl_server_number = 2
+--source include/rpl_stop_server.inc
+
+# Negative case 1 : Empty
+--let $wild_pattern=
+--let $expect_abort= yes
+--let $reason= Cannot process empty pattern
+--source extra/rpl_tests/rpl_enable_event_cmd_line_validation.inc
+
+# Negative case 2 : No DB
+--let $wild_pattern=.event%
+--let $expect_abort= yes
+--let $reason= The pattern doesn't follow the format <db_pattern>.<event_pattern>
+--source extra/rpl_tests/rpl_enable_event_cmd_line_validation.inc
+
+# Negative case 3 : No Event
+--let $wild_pattern=db%.
+--let $expect_abort= yes
+--let $reason= The pattern doesn't follow the format <db_pattern>.<event_pattern>
+--source extra/rpl_tests/rpl_enable_event_cmd_line_validation.inc
+
+# Negative case 4 : No dot separator
+--let $wild_pattern=event
+--let $expect_abort= yes
+--let $reason= The pattern doesn't follow the format <db_pattern>.<event_pattern>
+--source extra/rpl_tests/rpl_enable_event_cmd_line_validation.inc
+
+# Negative case 5 : Multiple dots
+--let $wild_pattern=db%.event%.random
+--let $expect_abort= yes
+--let $reason= The pattern has multiple dot characters. Please restart with format <db_pattern>.<event_pattern>
+--source extra/rpl_tests/rpl_enable_event_cmd_line_validation.inc
+
+# Negative case 6 : Channel separator
+--let $wild_pattern=channel:db%.event%
+--let $expect_abort= yes
+--let $reason= This server doesn't support per-channel --replica-enable-event feature
+--source extra/rpl_tests/rpl_enable_event_cmd_line_validation.inc
+
+# Positiv case 1 : DB.EVENT
+--let $wild_pattern=db%.event1%
+--let $expect_abort= no
+--source extra/rpl_tests/rpl_enable_event_cmd_line_validation.inc
+
+# Positiv case 2 : DB.EVENT
+--let $wild_pattern=d%b%.%event%
+--let $expect_abort= no
+--source extra/rpl_tests/rpl_enable_event_cmd_line_validation.inc
+
+# Restart the replica server and start the applier thread.
+--let $rpl_server_number= 2
+--let $rpl_server_parameters= --replica-enable-event=db%.event1% --replica-enable-event=d%b%.%event%
```

Comment:
does any other mysqld option work like this? I mean, used multiple times and all occurrences are accumulated.
Can we do a comma-separated list instead?

Reply:
All `--replicate-(do|ignore)-(db|table)` options work the same way. https://github.com/percona/percona-server/blob/8.0/sql/mysqld.cc#L8820-L8884.

Reply:
I see. MySQL behaviour is a bit inconsistent here. For some options, it would take the last value only. for example, sort_buffer_size or innodb_buffer_pool_size if passed multiple times, only the last one is considered.

Ok to match the other replicate-* options and keep this one as is